meta-dts-distro/recipes-dts/dts/dasharo-deploy/dasharo-deploy

```diff
@@ -617,6 +574,128 @@ install() {
   fi
 }
 
+update_ec() {
+  echo "Downloading EC firmware update..."
+  curl --progress-bar -f "$EC_LINK" -o "$EC_UPDATE_FILE"
+  error_check "Cannot access $FW_STORE_URL. Please check your internet connection"
+  echo "Checking sha256sum..."
+  sha256sum --check <(echo "$EC_HASH")
+  error_check "Failed to download EC firmware update"
+
+  echo "Updating EC..."
+  system76_ectool flash "$EC_UPDATE_FILE" &>> $ERR_LOG_FILE
+  error_check "Failed to update EC firmware"
+}
+
+compare_versions() {
+    # return 1 if ver2 > ver1
+    # return 0 otherwise
+
+    local ver1="$1"
+    local ver2="$2"
+
+    IFS='.' read -r -a arr_ver1 <<< "$ver1"
+    IFS='.' read -r -a arr_ver2 <<< "$ver2"
+
+    if [ ${arr_ver2[0]} -lt ${arr_ver1[0]} ]; then
+        return 0
+    fi
+
+    if [ ${arr_ver2[0]} -gt ${arr_ver1[0]} ]; then
+        return 1
+    fi
+
+    if [ ${arr_ver2[0]} -eq ${arr_ver1[0]} ]; then
+        if [ ${arr_ver2[1]} -lt ${arr_ver1[1]} ]; then
+            return 0
+        fi
+
+        if [ ${arr_ver2[1]} -gt ${arr_ver1[1]} ]; then
+            return 1
+        fi
+
+        if [ ${arr_ver2[1]} -eq ${arr_ver1[1]} ]; then
+            if [ ${arr_ver2[2]} -lt ${arr_ver1[2]} ]; then
+                return 0
+            fi
+
+            if [ ${arr_ver2[2]} -gt ${arr_ver1[2]} ]; then
+              return 1
+            fi
+        fi
+    fi
+}
+
+update() {
+  echo "Checking for the latest Dasharo update available..."
+  echo "Current Dasharo version: $DASHARO_VERSION"
+  echo "Latest available Dasharo version: $DASHARO_REL_VER"
+  compare_versions $DASHARO_VERSION $DASHARO_REL_VER
+  if [ $? -ne 1 ]; then
+    error_exit "No update available for your machine"
+  fi
+
+  BIOS_HASH=$BIOS_HASH_PROD
+  BIOS_LINK=$BIOS_LINK_PROD
+  if [ "$HAVE_EC" = "true" ]; then
+    EC_HASH=$EC_HASH_PROD
+    EC_LINK=$EC_LINK_PROD
+  fi
+
+  display_flashing_warning
+
+  echo "Downloading Dasharo firmware update..."
+  curl --progress-bar -f "$BIOS_LINK" -o $BIOS_UPDATE_FILE
+  error_check "Cannot access $FW_STORE_URL. Please check your internet connection"
+  echo "Checking sha256sum..."
+  sha256sum --check <(echo "$BIOS_HASH")
+  error_check "Failed to verify Dasharo firmware"
+
+  if [ "$HAVE_EC" = "true" ]; then
+    system76_ectool info >> $ERR_LOG_FILE 2>&1
+    if [ $? -ne 0 ]; then
```

Comment:
Can we use error_check here?

Reply:
fixed https://github.com/Dasharo/meta-dts/pull/29/commits/4d7244392eb909e4f4fa4709d32547c20cfce84a